### PR TITLE
Scouting - add track collection

### PR DESCRIPTION
**Pull request by swagata87**
These changes are meant for scouting event content only. The dimuon calo scouting HLT paths produce a track collection in order to compute track-isolation of muons, that track collection can be saved in event content, and can be used in analyses. Using edmEventSize, I obtained the size of the collection.
Branch Name   | Average Uncompressed Size (Bytes/Event) | Average Compressed Size (Bytes/Event) 
ScoutingTracks_hltScoutingTrackPacker__MYHLT.  |   1103.12   |   876.943

### CODE REVIEW

File: DataFormats/Scouting/src/classes_def.xml [hunk 1]
```diff
@@ -24,18 +24,23 @@
   <class name="ScoutingPhoton" ClassVersion="2">
       <version ClassVersion="2" checksum="2797410539"/>
   </class>
+  <class name="ScoutingTrack" ClassVersion="-1">
```
Comment (Dr15Jones) [inline DataFormats/Scouting/src/classes_def.xml]: The version number should be 2

File: HLTrigger/Muon/plugins/HLTScoutingTrackProducer.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#include "DataFormats/Scouting/interface/ScoutingVertex.h"
+
+#include "DataFormats/MuonReco/interface/MuonTrackLinks.h"
+#include "DataFormats/MuonReco/interface/MuonFwd.h"
+
+class HLTScoutingTrackProducer : public edm::global::EDProducer<> {
+
+    public:
+        explicit HLTScoutingTrackProducer(const edm::ParameterSet&);
+        ~HLTScoutingTrackProducer() override;
```
Comment (Dr15Jones) [inline HLTrigger/Muon/plugins/HLTScoutingTrackProducer.h]: There is no need to declare the destructor

File: HLTrigger/Muon/plugins/HLTScoutingTrackProducer.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+    private:
+        void produce(edm::StreamID sid, edm::Event & iEvent, edm::EventSetup const & setup)
+            const final;
+
+        const edm::EDGetTokenT<reco::TrackCollection> OtherTrackCollection_;
```
Comment (Dr15Jones) [inline HLTrigger/Muon/plugins/HLTScoutingTrackProducer.h]: CMS coding convention is to start member variables with a lower case letter

File: HLTrigger/Muon/plugins/HLTScoutingTrackProducer.cc [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+/**\class HLTScoutingTrackProducer HLTScoutingTrackProducer.cc HLTScoutingTrackProducer.cc
+
+Description: Producer for Scouting Tracks
+
+*/
+//
+
+#include "HLTScoutingTrackProducer.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "TMath.h"
```
Comment (Dr15Jones) [inline HLTrigger/Muon/plugins/HLTScoutingTrackProducer.cc]: Are the last two includes needed?

File: HLTrigger/Muon/plugins/HLTScoutingTrackProducer.h [hunk 5]
```diff
@@ -0,0 +1,62 @@
+#ifndef HLTScoutingTrackProducer_h
+#define HLTScoutingTrackProducer_h
+
```
Comment (Dr15Jones) [inline HLTrigger/Muon/plugins/HLTScoutingTrackProducer.h]: It is best to remove this file and put its content directly in the corresponding .cc file.

**Comment by Dr15Jones** [comment]
please test

**Comment by Martin-Grunewald** [comment]
We need a 10_1 PR first!

**Comment by Martin-Grunewald** [comment]
+1

**Comment by Dr15Jones** [comment]
+1

**Comment by Martin-Grunewald** [comment]
backport of #22418

**Comment by fabiocos** [comment]
+1

**Pull request merged**